my_account_logged_in.xml weird landscape text truncate

shkshreyas opened the pull request:
<!--- Please fill the necessary details below -->
## Purpose / Description
On Landscape the images look weird and text truncate issue
`solved the problem
<h3>AnkiWeb account</h3>
previous problem 
<img src="https://i.ibb.co/ygrjh9K/Whats-App-Image-2025-01-14-at-21-27-07-15b0b224.jpg" alt="Portrait View" width="400" />
Solved 
<img src="https://i.ibb.co/k8tCLpQ/Whats-App-Image-2025-01-14-at-21-27-07-18b35b16.jpg" alt="Portrait View" width="400"/>
<img src="https://i.ibb.co/W2CL4qV/Whats-App-Image-2025-01-14-at-21-27-07-7b8455be.jpg" alt="Portrait View" width="400"/>
<img src="https://i.ibb.co/3762Bq5/Whats-App-Image-2025-01-14-at-21-27-08-434a650e.jpg" alt="Portrait View" width="400"/>
 Added Scroll view in landscape mode also modified layouts and text distorted part 

## Fixes
* Fixes #17312

## Approach
_How does this change address the problem?_
As you can see the above screenshots I have added scroll view layouts also modified some part of code so that on landscape it looks better and scrollable
## How Has This Been Tested?
It is tested both by emulator with 34 level and hardware in realme 6 Android 11

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [yes ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ yes] You have commented your code, particularly in hard-to-understand areas
- [ yes] You have performed a self-review of your own code
- [ yes] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ yes] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

shkshreyas commented:
I think this error is solved

david-allison commented:
Hi @shkshreyas Is there a relevant issue, or can you list before/after screenshots?

shkshreyas commented:
have you accepted this. so close the issue ?

shkshreyas commented:
> Hi @shkshreyas Is there a relevant issue, or can you list before/after screenshots?

`solved the problem
<h3>AnkiWeb account</h3>
previous problem 
<img src="https://i.ibb.co/ygrjh9K/Whats-App-Image-2025-01-14-at-21-27-07-15b0b224.jpg" alt="Portrait View" width="400" />
Solved 
<img src="https://i.ibb.co/k8tCLpQ/Whats-App-Image-2025-01-14-at-21-27-07-18b35b16.jpg" alt="Portrait View" width="400"/>
<img src="https://i.ibb.co/W2CL4qV/Whats-App-Image-2025-01-14-at-21-27-07-7b8455be.jpg" alt="Portrait View" width="400"/>
<img src="https://i.ibb.co/3762Bq5/Whats-App-Image-2025-01-14-at-21-27-08-434a650e.jpg" alt="Portrait View" width="400"/>
 Added Scroll view in landscape mode also modified layouts and text distorted part

david-allison commented:
> have you accepted this. so close the issue ?

Sorry, I don't understand. Is there an issue in our repository which this solves, or does this PR both define and fix the issue?

shkshreyas commented:
> > have you accepted this. so close the issue ?
> 
> Sorry, I don't understand. Is there an issue in our repository which this solves, or does this PR both define and fix the issue?

Issue: UI Bug on AnkiWeb Login Screen in Landscape Mode (#17312)
Problem Description
In the landscape mode of the AnkiWeb Login screen, there were two key issues affecting usability and design:

Text Misplacement: The "Logged in as" label appeared at the top of the navbar, making it visually inconsistent.
Text Truncation: Some text elements were cut off, leading to readability issues.
This behavior occurred only in landscape mode and did not affect the desktop version of the application.

What’s Fixed
The following improvements were implemented to address the issue:

Added ScrollView:
In landscape mode, if the total page content exceeds the visible area, users can now scroll seamlessly to access hidden elements.
Adjusted "Logged in as" Position:
The label is now correctly placed below the navbar, ensuring a clean and consistent layout.
Resolved Text Truncation:
Truncated text elements have been fixed to display properly across all screen sizes and orientations.
Screenshots
Before (Landscape):
"Logged in as" text improperly positioned at the top of the navbar.
Partial text truncation issues.
After (Landscape):
Proper positioning of "Logged in as" below the navbar.
ScrollView enabled for overflowing content.
Text truncation resolved, enhancing readability and aesthetics.
How It’s Done
Introduced ScrollView in the XML layout to ensure content is scrollable in landscape mode.
Refined layout XML files to reposition the "Logged in as" label dynamically for better UI consistency.
Fixed text size and truncation issues using appropriate layout parameters.
Testing and Accessibility
Verified changes on both portrait and landscape orientations.
Passed tests using [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor) to ensure compatibility.
Next Steps
The fix is complete and ready for review.
Feedback and suggestions for further UI optimizations are welcome.

mikehardy commented on gradle/libs.versions.toml:
```diff
@@ -18,7 +18,7 @@ ktlint = '1.5.0'
 # Old changelogs - See 'Table of Contents' in sidebar
 # https://developer.android.com/build/releases/past-releases
-androidGradlePlugin = "8.7.3"
+androidGradlePlugin = "8.8.0"
```
don't bump dependencies in PRs please, we handle those via dependabot

shkshreyas replied:
> don't bump dependencies in PRs please, we handle those via dependabot

corrected the version downgraded

mikehardy commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,92 +7,97 @@
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/remove_account_button"
+                    app:backgroundTint="@color/material_red_500"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:minWidth="200dp"
+                    android:layout_gravity="center"
+                    android:text="@string/remove_account"
+                    android:layout_margin="@dimen/content_vertical_padding" />
```
changing the close tags to be on same line as last attribute contributes lines to the diff that aren't making any changes, which wastes reviewer time, please don't change the formatting unnecessarily

shkshreyas replied:
> changing the close tags to be on same line as last attribute contributes lines to the diff that aren't making any changes, which wastes reviewer time, please don't change the formatting unnecessarily

check the section after login in landscape mode then you will see the difference.

mikehardy replied:
I marked up the specific unnecessary changes I meant - each one is present in the diff only because you chose to alter the position of the element closure

That inflates the diff which wastes reviewer time, I request you do not do that

shkshreyas replied:
Ok. Have you accepted the pull request?

mikehardy replied:
> Ok. Have you accepted the pull request?

That's not how this works, I don't accept pull requests until the diff is clean and solves a real problem.

mikehardy commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,92 +7,97 @@
-                android:text="@string/remove_account"
-                android:layout_margin="@dimen/content_vertical_padding"
-                />
+                <com.ichi2.ui.FixedTextView
+                    android:id="@+id/username_logged_in"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textSize="@dimen/title_textsize"
+                    android:textStyle="bold"
+                    android:layout_margin="@dimen/content_vertical_padding"
+                    tools:text="user@example.com" />
```
unnecessary change

mikehardy commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,92 +7,97 @@
-                android:singleLine="true"
-                android:text="@string/help_title_privacy" />
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/logout_button"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:minWidth="200dp"
+                    android:layout_gravity="center"
+                    android:text="@string/log_out"
+                    android:layout_margin="@dimen/content_vertical_padding" />
```
unnecessary change

mikehardy replied:
Please address my comments regarding spacing. You are ignoring them. I mean for you to do the XML closures like this

```suggestion
                    android:layout_margin="@dimen/content_vertical_padding"
                     />
```

mikehardy commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,92 +7,97 @@
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/remove_account_button"
+                    app:backgroundTint="@color/material_red_500"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:minWidth="200dp"
+                    android:layout_gravity="center"
+                    android:text="@string/remove_account"
+                    android:layout_margin="@dimen/content_vertical_padding" />
```
unnecessary change

david-allison commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,84 +7,95 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:fitsSystemWindows="true">
+
```
revert please

shkshreyas replied:
done

david-allison commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -7,84 +7,95 @@
         android:foregroundGravity="center">
 
-        <include layout="@layout/toolbar" />
+        <include layout="@layout/toolbar"
```
revert please

shkshreyas replied:
> revert please

done

david-allison reviewed (commented):
Thanks for the PR!

* Could you reduce it to a minimal diff
* But before that, the AnkiDroid image in the vertical phone layout seems truncated. Are there still visual issues?

shkshreyas commented:
> Thanks for the PR!
> 
> * Could you reduce it to a minimal diff
> * But before that, the AnkiDroid image in the vertical phone layout seems truncated. Are there still visual issues?

Did as much as I could, but the code addition resulted in a large diff.

shkshreyas commented:
> Thanks for the PR!
> 
> * Could you reduce it to a minimal diff
> * But before that, the AnkiDroid image in the vertical phone layout seems truncated. Are there still visual issues?

![WhatsApp Image 2025-01-17 at 16 26 50_32d7d28d](https://github.com/user-attachments/assets/93678f04-efdc-483f-9b32-9527f5dbe9f5)
![WhatsApp Image 2025-01-17 at 16 26 50_37b76f6e](https://github.com/user-attachments/assets/bd46df08-f412-484f-992e-e0b5b8638954)
![WhatsApp Image 2025-01-17 at 16 26 51_eec63b55](https://github.com/user-attachments/assets/108342d9-c43e-4562-ba26-1b1008feb346)

shkshreyas commented:
![Screenshot_2025-01-17-19-23-44-00_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/926c82f9-97c8-4545-acc0-9f5c96c13475)
I have changed the line 10. Also the closing tags are same as the previous code with main Apk in anki droid github

mikehardy commented:
> Did as much as I could, but the code addition resulted in a large diff.

That makes sense, given the addition of a ScrollView necessarily changes the indent everywhere.

However, the diff is *finally* *finally* small when looking at it with whitespace-ignored, which was the goal of my previous comments as well:

https://github.com/ankidroid/Anki-Android/pull/17818/files?diff=unified&w=1

Notice how you can now clearly see what the *actual* changes are? This is critical for a reviewer of any of your PRs, and please always keep in mind when proposing changes to open source software that maintainer time, and specifically the time maintainers have to review PRs, is usually the most precious time of all in high-traffic repos. We get *lots* of PRs, but we do not have a lot of time to review them. So make your diffs small by default if possible, without requiring a bunch of requests to do so

mikehardy reviewed (approved):
I'm finally able to read the diff 😆  and it seems good - appears to be a good change

shkshreyas commented:
@david-allison  I think I have solved the issue?

david-allison commented on AnkiDroid/src/main/res/layout/my_account_logged_in.xml:
```diff
@@ -15,76 +15,82 @@
-                />
+                <ImageView
+                    android:id="@+id/login_logo"
+                    android:layout_width="330dp"
```
unusual change, but sure

david-allison reviewed (approved):
Looks great. Cheers!

david-allison commented:
@shkshreyas I've squashed your commit series. In future, please rebase + squash when making changes, rather than introducing merge commits
